lifecycle/lifecycle/job/async_call.py

```diff
@@ -0,0 +1,73 @@
+from racetrack_client.log.errors import EntityNotFound
+from racetrack_client.utils.time import timestamp_to_datetime
+from racetrack_commons.entities.dto import AsyncJobCallDto
+from lifecycle.django.registry import models
+
+
+def get_async_job_call(call_id: str) -> models.AsyncJobCall:
+    try:
+        return models.AsyncJobCall.objects.get(id=call_id)
+    except models.AsyncJobCall.DoesNotExist:
+        raise EntityNotFound(f'Async job call with id "{call_id}" was not found')
+
+
+def save_async_job_call(dto: AsyncJobCallDto) -> models.AsyncJobCall:
+    changed = False
+    try:
+        model = get_async_job_call(dto.id)
+    except EntityNotFound:
+        model = models.AsyncJobCall(id=dto.id)
+        changed = True
+
+    new_started_at = timestamp_to_datetime(dto.started_at)
+    new_ended_at = timestamp_to_datetime(dto.ended_at) if dto.ended_at is not None else None
+    new_request_body: bytes = dto.request_body.encode()
+    new_response_body: bytes = dto.response_body.encode()
+
+    if model.status != dto.status \
+            or model.started_at != new_started_at \
+            or model.ended_at != new_ended_at \
+            or model.error != dto.error \
+            or model.job_name != dto.job_name \
+            or model.job_version != dto.job_version \
+            or model.job_path != dto.job_path \
+            or model.request_method != dto.request_method \
+            or model.request_url != dto.request_url \
+            or model.request_headers != dto.request_headers \
+            or model.request_body != new_request_body \
+            or model.response_status_code != dto.response_status_code \
+            or model.response_headers != dto.response_headers \
+            or model.response_body != new_response_body \
+            or model.attempts != dto.attempts \
+            or model.pub_instance_addr != dto.pub_instance_addr \
+            or model.retriable_error != dto.retriable_error:
+        changed = True
+    model.status = dto.status
+    model.started_at = new_started_at
+    model.ended_at = new_ended_at
+    model.error = dto.error
+    model.job_name = dto.job_name
+    model.job_version = dto.job_version
+    model.job_path = dto.job_path
+    model.request_method = dto.request_method
+    model.request_url = dto.request_url
+    model.request_headers = dto.request_headers
+    model.request_body = new_request_body
+    model.response_status_code = dto.response_status_code
+    model.response_headers = dto.response_headers
+    model.response_body = new_response_body
+    model.attempts = dto.attempts
+    model.pub_instance_addr = dto.pub_instance_addr
+    model.retriable_error = dto.retriable_error
```

Comment:
It is verbose indeed. Usually, it's one-to-one mapping (it converts JSON data into database fields). However, for some fields we need to do a basic transformation, like converting string into bytes or integer into datetime, and that makes it not really a repeatable, automated process.

Plus, I've noticed that Django ORM always makes an `UPDATE` statement on database record, even if nothing changed, so that's why I made this huge `if` check.

To make it more generic, I think I could move the definition of fields and mappers into some dictionary, but then I'd need to access the fields with [Python's `setattr`](https://docs.python.org/3/library/functions.html#setattr) by string name, which makes it more error-prone for future refactors in my opinion.